torch/_dynamo/side_effects.py

```diff
@@ -202,12 +205,22 @@ def load_attr(self, item, name, deleted_ok=False, check=False):
             assert self.is_attribute_mutation(item)
         result = self.store_attr_mutations[item][name]
         if not deleted_ok and isinstance(result, variables.DeletedVariable):
-            unimplemented("read deleted attribute")
+            unimplemented_v2(
+                gb_type="Attempted to read a deleted variable",
+                context=f"item: {item}, name: {name}",
+                explanation="",
+                hints=[],
+            )
         return result
 
     def store_cell(self, cellvar, value):
         if cellvar.is_immutable():
-            unimplemented("Dynamo currently doesn't support writing to such cell")
+            unimplemented_v2(
+                gb_type="Write to immutable cell",
+                context=f"cellvar: {cellvar}, value: {value}",
+                explanation="Dynamo doesn't support writing to immutable cell variables.",
```

Comment:
Why are some cell variables immutable? Are they immutable in cpython or is this a Dynamo concept?

Reply:
cc @StrongerXi this looks like a Dynamo concept - we don't support mutating sourceless cells?

Reply:
If this is a Dynamo concept, then we need to explain these to users differently

Reply:
Oh yeah super edge case, I think the "right solution" here is for us Dynamo developers to audit the sources of this, and either fix them or provide a workaround to user. I should've dropped a note to this code block here. Maybe we should just create an issue.

https://github.com/pytorch/pytorch/blob/1c26a95ae3fcb347ebdf36ef7719e6d819d0487f/torch/_dynamo/variables/functions.py#L336-L343